pyk/src/pyk/ktool/kompile.py

```diff
@@ -43,13 +60,25 @@ def kompile(
 
 def _build_arg_list(
     *,
+    main_module: Optional[str],
+    syntax_module: Optional[str],
     backend: Optional[KompileBackend],
     output_dir: Optional[Path],
     include_dirs: Iterable[Path],
-    emit_json: bool,
+    md_selector: Optional[str],
+    hook_namespaces: Iterable[str],
+    emit_json,
+    concrete_rules: Iterable[str],
+    additional_args: Iterable[str]
 ) -> List[str]:
     args = []
 
+    if main_module:
+        args.extend(['--main-module', main_module])
+
+    if syntax_module:
+        args.extend(['--syntax-module', syntax_module])
+
```

Comment:
Do we want to default this to the `main_module` if it's not provided? That way we get less warnings from the frontend, but get the same default behavior?

Reply:
I'd prefer letting the frontend deal with defaults, rather than duplicating that logic. If we feel that the warning is an overkill we should change the behaviour in the frontend rather than having an opinion here.

Reply:
It's also easy enough for the client to suppress the warning by supplying a syntax module.